wordgenerator.py

```diff
@@ -0,0 +1,110 @@
+import pygame
+import random
+import sys
+import math
+from copy import deepcopy
+
+CONST_FILENAME = "american-english.txt"  
+
+'''
+Opens the file CONST_FILENAME and reads entire file to word_array list. Shuffles the array and then sorts according to length
+'''
+class WordGenerator():
+    index_locations = []
+    index_prime_locations = []
+    word_array = []
+    number_of_lengths = 0
+    smallest_length = []
+
+    def __init__(self):
+        with open(CONST_FILENAME) as infile:
+            for line in infile:
+                self.word_array.append(line.rstrip('\n'))
+        random.shuffle(self.word_array) 
+        self.word_array.sort(key=len)
+        self.split_list(self.word_array)
+
+    '''
+    Will populate the list index_locations on where each new length starts
+    For example, index_locations[3] will give you the location of the first word with length 3
+    index_locations[4] will give you location of first word with length 4, etc.
+    index_prime_locations will be incremented and used to keep track of which words have been given out.
+    '''
+    def split_list(self, list):
+        self.number_of_lengths = len(list[len(list)-1]) - len(list[0]) 
+        self.smallest_length = len(list[0])
+        #For lengths that are smaller than the smallest word, we point them to index 0 in the array
+        for x in range (0, self.smallest_length):
+            self.index_locations.append(0)
+        for x in range (self.smallest_length, self.number_of_lengths):
+            self.index_locations.append(self.locate_first_length(x))
+        self.index_prime_locations = deepcopy(self.index_locations)
+
+    '''
+    log(n) binary search to find the first length
+    '''
+    def locate_first_length(self, requested_length):
+        found = False
+        index = math.ceil(len(self.word_array)/2)
+        delta = math.ceil(len(self.word_array)/4)
+        while True:
+            if self.check_if_first(index, requested_length):
+                break
+            if len(self.word_array[index]) >= requested_length:
+                index -= delta
+            else:
+                index += delta 
+            if index < 0:
+                index = 0
+            delta = math.ceil(delta/2)
+        return index
+
+    def check_if_first(self, index, requested_length):
+        if len(self.word_array[index]) != requested_length:
+            return False
+        if index == 0:
+            return True
+        return len(self.word_array[index]) != len(self.word_array[index-1])
+
+    '''
+    If you want words with length 4-6 (inclusive), request_word(4,3)
+    If you want words with length of 2, request_word(2,1)
+    '''
+    def request_word(self, index, number_of_words):
```

Comment:
Reading from line 70-73 I'm confused at the parameter names. I don't feel like `index` and `number_of_words` really reflect the use case from the above comments. Looks more like `shortest_length` and `length_range` from the comments.